src/Uno.UI/DataBinding/BindingPropertyHelper.cs

```diff
@@ -1411,6 +1411,48 @@ private static object UnsetValueGetter(object unused)
 		/// <remarks>This method needs to be aligned with the symbols query in BindableTypeProvidersSourceGenerator.</remarks>
 		private static bool IsValidMetadataProviderType(Type type)
 			=> type.IsPublic && type.IsClass;
+
+		private class GetPropertyTypeKey
```

Comment:
I think most of this should be a `readonly record struct` with only the hash caching part exposed.

Reply:
If we will follow https://github.com/unoplatform/uno/pull/13963#discussion_r1357153836 to reduce allocations of this class, then moving to a struct will re-introduce the allocations as boxing allocations since Hashtable deals with `object`s everywhere.

Reply:
Also we wouldn't be able to mutate it.